### PR TITLE
fix: always represent node IDs with 16 hex digits

### PR DESCRIPTION
**Pull request by towynlin**
Add leading zeros if necessary

### CODE REVIEW

File: src/apps/bm_soft_module/user_code/user_code.cpp [hunk 1]
```diff
@@ -206,7 +206,7 @@ static bool BmSoftWatchdogHandler(void *arg) {
 
 static int createBmSoftDataTopic(void) {
   int topiclen = snprintf(bmSoftTopic, BM_TOPIC_MAX_LEN,
-                          "sensor/%" PRIx64 "/sofar/bm_soft_temp", getNodeId());
+                          "sensor/%016" PRIx64 "/sofar/bm_soft_temp", getNodeId());
```
Comment (towynlin) [inline src/apps/bm_soft_module/user_code/user_code.cpp:209]: This PR mostly changes logs. However, this is one of the types of more load-bearing changes — the pub-sub sensor reading topics contain node IDs, and we want to make sure they match on both sides, always containing leading zeros if necessary to reach 16 hex digits.

This is the topic on which SOFT modules publish.

File: src/apps/bridge/sensor_drivers/aanderaaSensor.cpp [hunk 2]
```diff
@@ -20,7 +20,8 @@ bool AanderaaSensor::subscribe() {
   bool rval = false;
   char *sub = static_cast<char *>(pvPortMalloc(BM_TOPIC_MAX_LEN));
   configASSERT(sub);
-  int topic_strlen = snprintf(sub, BM_TOPIC_MAX_LEN, "sensor/%" PRIx64 "%s", node_id, subtag);
+  int topic_strlen =
+      snprintf(sub, BM_TOPIC_MAX_LEN, "sensor/%016" PRIx64 "%s", node_id, subtag);
```
Comment (towynlin) [inline src/apps/bridge/sensor_drivers/aanderaaSensor.cpp:24]: Current meter subscription on the bridge

File: src/apps/bridge/sensor_drivers/rbrCodaSensor.cpp [hunk 3]
```diff
@@ -21,7 +21,8 @@ bool RbrCodaSensor::subscribe() {
   bool rval = false;
   char *sub = static_cast<char *>(pvPortMalloc(BM_TOPIC_MAX_LEN));
   configASSERT(sub);
-  int topic_strlen = snprintf(sub, BM_TOPIC_MAX_LEN, "sensor/%" PRIx64 "%s", node_id, subtag);
+  int topic_strlen =
+      snprintf(sub, BM_TOPIC_MAX_LEN, "sensor/%016" PRIx64 "%s", node_id, subtag);
```
Comment (towynlin) [inline src/apps/bridge/sensor_drivers/rbrCodaSensor.cpp:25]: RBR sensor reading subscription on the bridge

File: src/apps/bridge/sensor_drivers/softSensor.cpp [hunk 4]
```diff
@@ -20,7 +20,8 @@ bool SoftSensor::subscribe() {
   bool rval = false;
   char *sub = static_cast<char *>(pvPortMalloc(BM_TOPIC_MAX_LEN));
   configASSERT(sub);
-  int topic_strlen = snprintf(sub, BM_TOPIC_MAX_LEN, "sensor/%" PRIx64 "%s", node_id, subtag);
+  int topic_strlen =
+      snprintf(sub, BM_TOPIC_MAX_LEN, "sensor/%016" PRIx64 "%s", node_id, subtag);
```
Comment (towynlin) [inline src/apps/bridge/sensor_drivers/softSensor.cpp:24]: SOFT module subscription on the bridge

File: src/apps/bristleback_apps/aanderaa/user_code/user_code.cpp [hunk 5]
```diff
@@ -205,7 +205,8 @@ static bool aanderaaSensorWatchdogHandler(void *arg) {
 }
 
 static int createAanderaaDataTopic(void) {
-  int topiclen = snprintf(aanderaaTopic, BM_TOPIC_MAX_LEN, "sensor/%" PRIx64 "/sofar/aanderaa", getNodeId());
+  int topiclen = snprintf(aanderaaTopic, BM_TOPIC_MAX_LEN,
+                          "sensor/%016" PRIx64 "/sofar/aanderaa", getNodeId());
```
Comment (towynlin) [inline src/apps/bristleback_apps/aanderaa/user_code/user_code.cpp:213]: Current meter publish topic

File: src/apps/bristleback_apps/bm_rbr/user_code/user_code.cpp [hunk 6]
```diff
@@ -76,8 +76,8 @@ static bool BmRbrWatchdogHandler(void *arg) {
 }
 
 static int createBmRbrDataTopic(void) {
-  int topiclen = snprintf(bmRbrTopic, BM_TOPIC_MAX_LEN, "sensor/%" PRIx64 "/sofar/bm_rbr_data",
-                          getNodeId());
+  int topiclen = snprintf(bmRbrTopic, BM_TOPIC_MAX_LEN,
+                          "sensor/%016" PRIx64 "/sofar/bm_rbr_data", getNodeId());
```
Comment (towynlin) [inline src/apps/bristleback_apps/bm_rbr/user_code/user_code.cpp:80]: RBR publish topic

File: src/lib/middleware/services/config_cbor_map_service.cpp [hunk 7]
```diff
@@ -21,8 +21,8 @@ void config_cbor_map_service_init(cfg::Configuration& hw_cfg, cfg::Configuration
   _hw_config = &hw_cfg;
   _usr_config = &user_config;
   static char config_map_str[BM_SERVICE_MAX_SERVICE_STRLEN];
-  size_t topic_strlen = snprintf(config_map_str, sizeof(config_map_str),
-                                 "%" PRIx64 "%s", getNodeId(), CONFIG_MAP_SUFFIX);
+  size_t topic_strlen = snprintf(config_map_str, sizeof(config_map_str), "%016" PRIx64 "%s",
+                                 getNodeId(), CONFIG_MAP_SUFFIX);
```
Comment (towynlin) [inline src/lib/middleware/services/config_cbor_map_service.cpp:25]: config cbor map service topic

File: src/lib/middleware/services/echo_service.cpp [hunk 8]
```diff
@@ -21,8 +21,8 @@ static bool echo_service_handler (size_t service_strlen,
  */
 void echo_service_init(void) {
     static char echo_service_str[BM_SERVICE_MAX_SERVICE_STRLEN];
-    size_t topic_strlen = snprintf(echo_service_str, sizeof(echo_service_str), "%" PRIx64 "%s", getNodeId(),
-            ECHO_SERVICE_SUFFIX);
+    size_t topic_strlen = snprintf(echo_service_str, sizeof(echo_service_str),
+                                   "%016" PRIx64 "%s", getNodeId(), ECHO_SERVICE_SUFFIX);
```
Comment (towynlin) [inline src/lib/middleware/services/echo_service.cpp:25]: echo service topic

File: src/lib/middleware/services/sys_info_service.cpp [hunk 9]
```diff
@@ -27,7 +27,7 @@ void sys_info_service_init(Configuration &sys_config) {
   _sys_config = &sys_config;
   static char sys_info_service_str[BM_SERVICE_MAX_SERVICE_STRLEN];
   size_t topic_strlen = snprintf(sys_info_service_str, sizeof(sys_info_service_str),
-                                 "%" PRIx64 "%s", getNodeId(), SYS_INFO_SERVICE_SUFFIX);
+                                 "%016" PRIx64 "%s", getNodeId(), SYS_INFO_SERVICE_SUFFIX);
```
Comment (towynlin) [inline src/lib/middleware/services/sys_info_service.cpp:30]: sys info service topic

**Review by russelldeguzman — APPROVED** [review]

**Review by victorsowa12 — APPROVED** [review]
🤘

**Pull request merged**